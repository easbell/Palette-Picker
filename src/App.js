import React, { Component } from 'react';
import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      palettes: []
    }
  }

  componentDidMount = async () => {
    const url = process.env.REACT_APP_BACKEND_URL + '/api/v1/projects'
    const response = await fetch(url)
    const data = response.json()
    console.log(data)
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>Palette Picker</h1>
          {/* generate button */}
          {/* projects button */}
        </header>
        {/* random palette */}
        {/* save palette */}
      </div>
    )
  }
}

export default App;
