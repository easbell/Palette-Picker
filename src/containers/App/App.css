.App {
  text-align: center;
  font-family: 'Roboto', sans-serif;
  padding-bottom: 10px;
}

.header {
  background-color: #333;
  display: flex;
  align-items: center;
  padding: 20px 40px;
}

.logo {
  font-family: 'Pacifico', cursive;
  color: #fff;
  font-size: 1.5rem;
  margin: 0px;
}

.links {
  color: #999;
  font-size: 1rem;
  font-weight: bold;
  padding-left: 40px;
  margin: 0px;
}

.links:hover {
  color: #ea4c88;
  transition: all .2s ease;
}
