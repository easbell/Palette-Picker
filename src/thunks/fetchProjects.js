import { isLoading, setProjects, hasErrored } from '../actions'

export const fetchProjects = () => {
  const url = process.env.REACT_APP_BACKEND_URL + 'api/v1/projects';
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const projects = await response.json()
      dispatch(isLoading(false))
      dispatch(setProjects(projects))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}