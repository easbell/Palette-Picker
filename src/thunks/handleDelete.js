import { isLoading, hasErrored } from '../actions'

export const handleDelete = (url, action, items, id) => {
  return async (dispatch) => {
    dispatch(isLoading(true))
    try {
      const response = await fetch(url, {
        method: 'DELETE',
        body: JSON.stringify({id: id}),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      const msg = await response.json()
      console.log(msg)
      dispatch(isLoading(false))
      dispatch(action(items))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}