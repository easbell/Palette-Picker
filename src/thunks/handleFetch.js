import { isLoading, hasErrored } from '../actions'

export const handleFetch = (url, action, options) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, options)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const data = await response.json()
      dispatch(isLoading(false))
      dispatch(action(data))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}

