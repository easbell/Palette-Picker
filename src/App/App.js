import React, { Component } from 'react';
import './App.css';
import Palette from '../Palette/Palette'

export class App extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      palettes: []
    }
  }

  componentDidMount = async () => {
    const url = process.env.REACT_APP_BACKEND_URL + 'api/v1/projects'
    const response = await fetch(url)
    const data = await response.json()
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>Palette Picker</h1>
          {/* generate button */}
          {/* projects button */}
        </header>
        <Palette />
        <div>
          {/* save palette */}
        </div>
      </div>
    )
  }
}

export default App;
