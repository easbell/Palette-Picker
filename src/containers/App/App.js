import React, { Component } from 'react';
import './App.css';
import Palette from '../../Palette/Palette';
import { handleFetch } from '../../thunks/handleFetch';
import { connect } from 'react-redux';
import { setProjects, setPalettes } from '../../actions';
import { Route, Switch, Link, NavLink, withRouter } from 'react-router-dom'
import AllProjects from '../../containers/AllProjects/AllProjects'
import EditProject from '../EditProject/EditProject'

export class App extends Component {
  componentDidMount = () => {
  const allProjects = process.env.REACT_APP_BACKEND_URL + 'api/v1/projects';
  const allPalettes = process.env.REACT_APP_BACKEND_URL + 'api/v1/palettes';
  this.props.handleFetch(allProjects, setProjects)
  this.props.handleFetch(allPalettes, setPalettes)
  }

  findProject = ({ match }) => {
    const foundProject = this.props.projects.find(project => project.id == match.params.id)
    
    if(!foundProject) {
      return '404 Project not found'
    } else {
      return <EditProject foundProject={foundProject} />
    }
  }

  render() {
    return (
      <div className="App">
        <header>
          <Link to='/'>
            <h1>Palette Picker</h1>
          </Link>
          <NavLink
            exact to='/my-projects'
          >
            My Projects
          </NavLink>
        </header>
        <div>
          <Switch>
            <Route
              exact path='/'
              render={() => <Palette />}
            />
            <Route 
              exact path='/my-projects'
              render={() => <AllProjects />}
            />
            <Route 
              exact path='/my-projects/:id'
              render={this.findProject}
            />
          </Switch>
        </div>
        
        <div>
          {/* save palette */}
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  projects: state.projects
})

export const mapDispatchToProps = (dispatch) => ({
  handleFetch: (url, action) => dispatch(handleFetch(url, action))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
