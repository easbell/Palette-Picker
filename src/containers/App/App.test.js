import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { handleFetch } from '../../thunks/handleFetch';

jest.mock('../../thunks/handleFetch');

describe('App', () => {
  let wrapper;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn()
    wrapper = shallow(<App handleFetch={mockFn}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFetch twice upon componentDidMount', () => {
    wrapper = shallow(<App handleFetch={mockFn}/>)
    wrapper.instance().componentDidMount();

    expect(mockFn).toHaveBeenCalled();
  });

  describe('mapStateToProps', () => {
    it('returns correct props', () => {
      const mockState = {
        projects: ['projects'],
        other: 'other'
      }
      const expectedProps = {
        projects: ['projects']
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with handleFetch action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = handleFetch();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.handleFetch();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});