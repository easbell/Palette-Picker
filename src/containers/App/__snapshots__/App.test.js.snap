// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header>
    <Link
      to="/"
    >
      <h1>
        Palette Picker
      </h1>
    </Link>
    <NavLink
      exact={true}
      to="/my-projects"
    >
      My Projects
    </NavLink>
  </header>
  <div>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/my-projects"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/my-projects/:id"
        render={[Function]}
      />
    </Switch>
  </div>
  <div />
</div>
`;
